cli/command/telemetry_utils.go

```diff
@@ -15,9 +15,10 @@ import (
 )
 
 // BaseMetricAttributes returns an attribute.Set containing attributes to attach to metrics/traces
-func BaseMetricAttributes(cmd *cobra.Command) attribute.Set {
+func BaseMetricAttributes(cmd *cobra.Command, cli *DockerCli) attribute.Set {
```

Comment:
I'd rather accept more attribute KeyValue pairs than the dockerCli instance - unless `IsTerminal()` can change at runtime.

Reply:
Sure, that sounds good to me. I guess it might, but then again I don't think the underlying implementation of `IsTerminal()` gets updated, and in any case if someone runs the CLI from the terminal and then redirects the output then we probably still would like to count it as a "terminal" invocation.

Reply:
What I'd really like to do is set this on the resource level earlier, but by the point the resource is created we don't have everything in place yet to check without doing a larger refactor which I was trying to avoid.

Reply:
It's a bit of cheating but you can use [cmd.Annotations](https://pkg.go.dev/github.com/spf13/cobra#Command). We use that in a few places: https://github.com/docker/cli/blob/5786f20687f334146a229f0954e968c8b43ab92a/cli-plugins/manager/cobra.go#L15-L40

Reply:
I'd also avoid passing the whole dockerCli struct here, as being a nice and fat struct containing lots of things it seems a bit irrelevant

Reply:
Cheated a little by passing a `Streams` interface instead, but also did some refactoring to prevent having to drill it down so much. Hopefully that's an acceptable compromise.